### PR TITLE
UX: Updated Custom network displayed for selectable in network selection

### LINKED ISSUE #22559: [Bug]: Custom network not displayed or selectable in network selection
### Describe the bug

Was testing with develop and noticed in commit 1e5e386b3ee6195a2e21e9842d493e7aea4efb9c that after I added a custom network that it was not displayed in the network selection drop down.  This is not the case in production with 11.7.3.

### Expected behavior

Once a custom network is added, it is presented and can be selected from the drop down.

### Screenshots/Recordings

11.7.3 production stable recording: https://recordit.co/tjxEqyZsuj
Commit 1e5e386b3ee6195a2e21e9842d493e7aea4efb9c in a flask build: https://recordit.co/6ASR0uVsPz

### Steps to reproduce

1. Have ganache running locally with `npx ganache-cli --fork https://mainnet.infura.io/v3/[YOUROWNINFURAID] --chainId 1`
2. Add it as a custom network with chainId=1
3. Select the newly added network
4. Change to another network
5. Note that you are no longer able to change back to custom network even though it remains present

### Error messages or log output

_No response_

### Version
Observed initially while testing against develop commit:
1e5e386b3ee6195a2e21e9842d493e7aea4efb9c

Can now confirm that 2f0f141b6e52061d6acd097f600f81bde9bb26a1 appears to be where this was introduced.

### Build type

Flask

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

Mac sonoma 14.2.1 with Chrome 117.0.5938.92

### PR DESCRIPTION
**Pull request by NidhiKJha**
This PR is to make sure if two networks with same id but different rpcUrl are there, list should show both of them

## **Related issues**

Fixes: #22559 

## **Manual testing steps**

1. Have ganache running locally with npx ganache-cli --fork https://mainnet.infura.io/v3/[YOUROWNINFURAID] --chainId 1
2. Add it as a custom network with chainId=1
3. Select the newly added network
4. Change to another network
5. Note that you are able to change back to see the custom network in the list and able to switch back.

### **After**

Screenshot 2024-01-29 at 6 07 35 PM

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/controllers/network-order.ts [hunk 1]
```diff
@@ -116,18 +124,31 @@ export class NetworkOrderController extends BaseController<
...
+      if (
+        !uniqueArray.some(
+          (uniqueItem) =>
+            uniqueItem.networkRpcUrl === item.networkRpcUrl &&
+            uniqueItem.networkId === item.networkId,
+        )
+      ) {
+        uniqueArray.push(item);
+      }
+    });
```
Comment (DDDDDanica) [inline app/scripts/controllers/network-order.ts:145]: Nit: try this function from lodash to filter duplicated elements 
```
const uniqueArray = combinedArray.uniqBy((network) => {
  return { networkRpcUrl: network.networkRpcUrl, networkId: network.networkId };
});
```

File: ui/selectors/selectors.js [hunk 2]
```diff
@@ -1251,6 +1251,7 @@ export function getSortedAnnouncementsToShow(state) {
 }
 
 export function getOrderedNetworksList(state) {
+  console.log(state.metamask.orderedNetworkList)
```
[the remark this answers is not in the document]
Comment (NidhiKJha) [reply ui/selectors/selectors.js]: Removed

File: ui/components/multichain/network-list-menu/network-list-menu.js [hunk 3]
```diff
@@ -96,18 +96,25 @@ export const NetworkListMenu = ({ onClose }) => {
...
+    });
+
+    // Sort nonTestNetworks based on the order in orderedNetworksList
+    const sortedNonTestNetworks = nonTestNetworks.sort((a, b) => {
+      const keyA = `${a.chainId}_${a.rpcUrl}`;
+      const keyB = `${b.chainId}_${b.rpcUrl}`;
+      return orderedIndexMap[keyA] - orderedIndexMap[keyB];
+    });
+
+    return sortedNonTestNetworks;
```
Comment (darkwing) [inline ui/components/multichain/network-list-menu/network-list-menu.js:112]: Can this data manipulation be moved to a selector?  It seems like a lot to be doing in side a UI component.

Comment (NidhiKJha) [reply ui/components/multichain/network-list-menu/network-list-menu.js:112]: @darkwing  I think we can do it in a follow up PR with the e2e test that we talked about today?

**Review by darkwing — APPROVED** [review]
Works wonderfully.  We should move some code to selectors but can be done in a follow up!

**Review by vthomas13 — APPROVED** [review]

**Pull request merged**